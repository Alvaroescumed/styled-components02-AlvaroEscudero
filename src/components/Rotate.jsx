import { keyframes, styled } from "styled-components";

// creamos los frames de la animacion
const rotate = keyframes`
    from{
        transform: rotate(0);
    }
    to{
        transform: rotate(360deg);
    }
`
// creamos la funcion
export const Rotate = styled.div`
    animation: ${rotate} 10s cubic-bezier(0.165, 0.84, 0.44, 1) infinite;
    margin-left: 50vw;
`