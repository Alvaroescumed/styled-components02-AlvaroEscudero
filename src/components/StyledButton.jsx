import styled, { css } from 'styled-components';

export const Button = styled.button`
    background-color: transparent;
    border: 2px solid blue;
    color: blue;
    padding:  10px 20px 10px 20px;
    border-radius: 20px;

    &:hover{
        background-color:  lightblue;
    }
    &:focus{
        background-color:  darkblue;
        color: white;
    }

    ${props => props.succes && css`
        border-color: green;
        color: green;

        &:hover{
            background-color: lightgreen;
        }
        &:focus{
            background-color: darkgreen;
        }
    `}
    ${props => props.error && css`
        border-color: #dc4b32;
        color: #dc4b32;

        &:hover{
            background-color: #f09b9a;
        }
        &:focus{
            background-color: #771616;
        }
    `}
    ${props => props.warning && css`
        border-color: #E9BA28;
        color: #E9BA28;

        &:hover{
            background-color: #f9e9bb;
        }
        &:focus{
            background-color: #af870e;
        }
    `}
    ${props => props.info && css`
        border-color: #982cbc;
        color: #982cbc;

        &:hover{
            background-color: #d0abf2;
        }
        &:focus{
            background-color: #3f156c;
        }
    `}



`;
