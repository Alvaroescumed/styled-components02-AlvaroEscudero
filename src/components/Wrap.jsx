import styled from "styled-components";

export const Wrap = styled.div`
    display: grid;
    grid-template-columns: repeat(5, 1fr);
    grid-column-gap: 20px;
    margin: 20px ;

`