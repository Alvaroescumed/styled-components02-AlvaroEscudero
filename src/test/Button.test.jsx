import { getByRole, render, screen } from "@testing-library/react";
import { Button } from "../components/StyledButton";
import { expect, test } from "vitest";

// testeamos que el componente renderice bien
test('Should render correctly', () => {
    render(<Button>test</Button>)
})

// testeamos que el boton en todas sus variantes concuerden con la snaphot 

test('Should match snapshou without props', () => {
    render(<Button>test</Button>)
    expect(screen.getByRole('button')).toMatchSnapshot()
})
test('Should match snapshou with succes prop', () => {
    render(<Button succes>test</Button>)
    expect(screen.getByRole('button')).toMatchSnapshot()
})
test('Should match snapshou with error prop', () => {
    render(<Button error>test</Button>)
    expect(screen.getByRole('button')).toMatchSnapshot()
})
test('Should match snapshou with info prop', () => {
    render(<Button info>test</Button>)
    expect(screen.getByRole('button')).toMatchSnapshot()
})
test('Should match snapshou with warning prop', () => {
    render(<Button warning>test</Button>)
    expect(screen.getByRole('button')).toMatchSnapshot()
})
