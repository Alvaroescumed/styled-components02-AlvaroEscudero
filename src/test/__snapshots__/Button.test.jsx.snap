// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should match snapshou with error prop 1`] = `
.c0 {
  background-color: transparent;
  border: 2px solid blue;
  color: blue;
  padding: 10px 20px 10px 20px;
  border-radius: 20px;
  border-color: #dc4b32;
  color: #dc4b32;
}

.c0:hover {
  background-color: lightblue;
}

.c0:focus {
  background-color: darkblue;
  color: white;
}

.c0:hover {
  background-color: #f09b9a;
}

.c0:focus {
  background-color: #771616;
}

<button
  class="c0"
>
  test
</button>
`;

exports[`Should match snapshou with info prop 1`] = `
.c0 {
  background-color: transparent;
  border: 2px solid blue;
  color: blue;
  padding: 10px 20px 10px 20px;
  border-radius: 20px;
  border-color: #982cbc;
  color: #982cbc;
}

.c0:hover {
  background-color: lightblue;
}

.c0:focus {
  background-color: darkblue;
  color: white;
}

.c0:hover {
  background-color: #d0abf2;
}

.c0:focus {
  background-color: #3f156c;
}

<button
  class="c0"
>
  test
</button>
`;

exports[`Should match snapshou with succes prop 1`] = `
.c0 {
  background-color: transparent;
  border: 2px solid blue;
  color: blue;
  padding: 10px 20px 10px 20px;
  border-radius: 20px;
  border-color: green;
  color: green;
}

.c0:hover {
  background-color: lightblue;
}

.c0:focus {
  background-color: darkblue;
  color: white;
}

.c0:hover {
  background-color: lightgreen;
}

.c0:focus {
  background-color: darkgreen;
}

<button
  class="c0"
>
  test
</button>
`;

exports[`Should match snapshou with warning prop 1`] = `
.c0 {
  background-color: transparent;
  border: 2px solid blue;
  color: blue;
  padding: 10px 20px 10px 20px;
  border-radius: 20px;
  border-color: #E9BA28;
  color: #E9BA28;
}

.c0:hover {
  background-color: lightblue;
}

.c0:focus {
  background-color: darkblue;
  color: white;
}

.c0:hover {
  background-color: #f9e9bb;
}

.c0:focus {
  background-color: #af870e;
}

<button
  class="c0"
>
  test
</button>
`;

exports[`Should match snapshou without props 1`] = `
.c0 {
  background-color: transparent;
  border: 2px solid blue;
  color: blue;
  padding: 10px 20px 10px 20px;
  border-radius: 20px;
}

.c0:hover {
  background-color: lightblue;
}

.c0:focus {
  background-color: darkblue;
  color: white;
}

<button
  class="c0"
>
  test
</button>
`;

exports[`Should match snapshou without props 2`] = `
.c0 {
  background-color: transparent;
  border: 2px solid blue;
  color: blue;
  padding: 10px 20px 10px 20px;
  border-radius: 20px;
}

.c0:hover {
  background-color: lightblue;
}

.c0:focus {
  background-color: darkblue;
  color: white;
}

<button
  class="c0"
>
  test
</button>
`;

exports[`Should match snapshou without props 3`] = `
.c0 {
  background-color: transparent;
  border: 2px solid blue;
  color: blue;
  padding: 10px 20px 10px 20px;
  border-radius: 20px;
}

.c0:hover {
  background-color: lightblue;
}

.c0:focus {
  background-color: darkblue;
  color: white;
}

<button
  class="c0"
>
  test
</button>
`;

exports[`Should match snapshou without props 4`] = `
.c0 {
  background-color: transparent;
  border: 2px solid blue;
  color: blue;
  padding: 10px 20px 10px 20px;
  border-radius: 20px;
}

.c0:hover {
  background-color: lightblue;
}

.c0:focus {
  background-color: darkblue;
  color: white;
}

<button
  class="c0"
>
  test
</button>
`;
